src/js/libcs/OpDrawing.js

```diff
@@ -106,6 +106,183 @@ evaluator.fillcircle$2 = function(args, modifs) {
+        // x, y vals of the center
+        var pt = [cen.value[0].value.real, cen.value[1].value.real];
+
+        // transform to canvas
+        var m = csport.drawingstate.matrix;
+        var xx = pt[0] * m.a - pt[1] * m.b + m.tx;
+        var yy = pt[0] * m.c - pt[1] * m.d - m.ty;
+
+
+        // check for counter clockwise drawing
+        var cclock = List.det3(a, b, c).value.real > 0;
+
+        // modifs handling
+        Render2D.handleModifs(modifs, Render2D.conicModifs);
+        var size = 4 / 2.5;
+        if (Render2D.size !== null)
+            size = Render2D.size;
+
+        csctx.save();
+        csctx.lineWidth = size;
+        csctx.strokeStyle = Render2D.lineColor;
+
+        // canvas circle radius 
+        var rad = arcDist.value.real * m.sdet;
+
+        csctx.beginPath();
+        csctx.translate(xx, yy);
+
+        // use the canvas arc function -- buggy in Chrome at least in Okt 15
+        var useArc = false;
+
+        if (useArc) {
+            csctx.arc(0, 0, arcDist.value.real * m.sdet, startAngle, endAngle, cclock);
+        } else {
+            var num = 500; // Number of segments
+
+            //  mod 2 pi in case startAngle > endAngle
+            if (startAngle > endAngle) endAngle = endAngle + Math.PI * 2;
+
+            // divide segments --  rotate counterclockwise if necessary
+            var ntler = !cclock ? (endAngle - startAngle) / num : -(2 * Math.PI - endAngle + startAngle) / num;
+
+            // drawing
+            csctx.moveTo(rad * Math.cos(startAngle), rad * Math.sin(startAngle));
+            var angl;
+            for (var ii = 0; ii <= num; ii++) {
+                angl = startAngle + ii * ntler;
+                csctx.lineTo(rad * Math.cos(angl), rad * Math.sin(angl));
+            }
+        }
+
+
+        if (df === "F" || false) {
+            csctx.fillStyle = Render2D.lineColor;
+            csctx.closePath();
+            csctx.fill();
+        }
+
+        if (df === "D") {
+            csctx.stroke();
+        }
+        csctx.restore();
+
+    } else { // segment case
+        var ptA = eval_helper.extractPoint(v1);
+        var ptB = eval_helper.extractPoint(v2);
+        var ptC = eval_helper.extractPoint(v3);
+
+        // dists
+        var dAB = (ptA.x - ptB.x) * (ptA.x - ptB.x) + (ptA.y - ptB.y) * (ptA.y - ptB.y);
+        var dAC = (ptA.x - ptC.x) * (ptA.x - ptC.x) + (ptA.y - ptC.y) * (ptA.y - ptC.y);
+        var dBC = (ptC.x - ptB.x) * (ptC.x - ptB.x) + (ptC.y - ptB.y) * (ptC.y - ptB.y);
+
+        // if 2 points are the same return nada;
+        if (dAB < 1e-12 || dAB < 1e-12 || dAB < 1e-12) return nada;
+
+        // drawing points
+        var pt1, pt2, pt3;
+
+        // check by dets if B is in the middle
+        var randP = List.realVector([10 * Math.random(), 10 * Math.random(), 10 * Math.random()]);
+        var detABP = List.det3(a, b, randP).value.real;
+        var detACP = List.det3(b, c, randP).value.real;
+        var Bmiddle = detABP * detACP > 0;
```

Comment:
I guess this works, too. For comparison, if you wanted to do this deterministically, you'd take `d` as the far point on the line, then compute the cross ratio `cr(a,c;b,d)` and check it's sign. You don't have to do it, I'm just offering this in case it's useful here or somewhere else.